src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/HandleCallAction.cs

```diff
@@ -80,7 +80,7 @@ internal partial struct HandleCallAction
             // TODO: niche APIs that we probably shouldn't even have added
             // We have to issue a warning, otherwise we could break the app without a warning.
             // This is not the ideal warning, but it's good enough for now.
-            _diagnosticContext.AddDiagnostic(DiagnosticId.UnrecognizedParameterInMethodCreateInstance, calledMethod.GetParameterDisplayName(1), calledMethod.GetDisplayName());
+            _diagnosticContext.AddDiagnostic(DiagnosticId.UnrecognizedParameterInMethodCreateInstance, calledMethod.GetParameter((ParameterIndex)(1 + (calledMethod.HasImplicitThis() ? 1 : 0))).GetDisplayName(), calledMethod.GetDisplayName());
```

Comment:
Actually it will not be equivalent. This method is called for both `Activator.CreateInstance(string assemblyName, string typeName)` which is static, as well as `AppDomain.CreateInstance(string assemblyName, string typeName)` which is instance method. So we need this "offset" thing to point to the right parameter (`typeName` in this case).